proposals/0002-root-signature-in-clang.md

```diff
@@ -402,23 +402,75 @@ registers are bound multiple times, or where there are multiple RootFlags
 entries, so subsequent stages should not assume that any given root signature in
 IR is valid.
 
+#### Metadata Schema
 
-> **Open Question**: what should the named metadata be?  There's options I
-> think...
+* RootConstant
+
+| ParameterType | ShaderVisibility | ShaderRegister | RegisterSpace | Num32BitValues |
+|---------------|------------------|----------------|---------------|----------------|
+| String        | int              | int            | int           | int            |
+
+
+* RootDescriptor
+
+| ParameterType | ShaderVisibility | ShaderRegister | RegisterSpace | RootDescriptorFlags |
+|---------------|------------------|----------------|---------------|---------------------|
+| String        | int              | int            | int           | int                 |
+
+
+* DescriptorRange
+
+| RangeType | NumDescriptors | BaseShaderRegister | RegisterSpace | DescriptorRangeFlags |
+|-----------|----------------|--------------------|---------------|----------------------|
+| String    | int            | int                | int           | int                  |
+
+
+* DescriptorTable
+
+| DescriptorRanges          |
+|---------------------------|
+| list of DescriptorRanges  |
+
+* StaticSampler
+  
+| Filter | AddressU | AddressV | AddressW | MipLODBias | MaxAnisotropy | ComparisonFunc | BorderColor | MinLOD | MaxLOD | ShaderRegister | RegisterSpace | ShaderVisibility |
+|--------|----------|----------|----------|------------|---------------|----------------|-------------|--------|--------|----------------|---------------|------------------|
+| int    | int      | int      | int      | float      | int           | int            | int         | float  | float  | int            | int           | int              |
+
+* RootSignature
+
+| RootFlag | Parameters                                      | StaticSamplers          |
+|----------|-------------------------------------------------|-------------------------|
+| int      | list of RootDescriptor/RootConstant/DescriptorTable | list of StaticSamplers |
+
+* Function RootSignature pair
+
+| EntryFunction | RootSignature |
+|----------|----------------|
+| Function | RootSignature  |
+
+* RootSignature table
+
+Named metadata with name "dx.rootsignatures".
+
+| Function RootSignature Pairs |
+|-------------------------------------|
+| list of Function RootSignature Pair |
```

Comment:
I think we should be more strict in the description of the metadata here.  I think when "list of Foo" appears, what it really means is "reference to a list of Foo", not that the list is somehow inlined into that metadata node.

Reply:
Change to `reference to list` and put Foo in Type detail.